packages/airnode-node/src/evm/verification/request-verification.ts

```diff
@@ -16,32 +16,34 @@ export function verifySponsorWallets<T>(
   requests: Request<T>[],
   masterHDNode: ethers.utils.HDNode
 ): LogsData<Request<T>[]> {
-  const logsWithVerifiedRequests: LogsData<Request<T>>[] = requests.map((request) => {
+  const logs: PendingLog[] = [];
+  const verifiedRequests: Request<T>[] = [];
+
+  requests.forEach(function (request) {
     if (request.status !== RequestStatus.Pending) {
       const message = `Sponsor wallet verification skipped for Request:${request.id} as it has status:${request.status}`;
       const log = logger.pend('DEBUG', message);
-      return [[log], request];
+      logs.push(log);
+      verifiedRequests.push(request);
+      return;
     }
 
     const expectedSponsorWalletAddress = wallet.deriveSponsorWallet(masterHDNode, request.sponsorAddress).address;
     if (request.sponsorWalletAddress !== expectedSponsorWalletAddress) {
+      // Drop request, but log error
       const message = `Invalid sponsor wallet:${request.sponsorWalletAddress} for Request:${request.id}. Expected:${expectedSponsorWalletAddress}`;
       const log = logger.pend('ERROR', message);
-      const updatedRequest: Request<T> = {
-        ...request,
-        status: RequestStatus.Ignored,
-        errorMessage: `${RequestErrorMessage.SponsorWalletInvalid}: ${request.sponsorWalletAddress}`,
-      };
-      return [[log], updatedRequest];
+      logs.push(log);
+      return;
```

Comment:
Note here that only the `log` is pushed and not the `request` i.e. the `request` is dropped.

Using two separate arrays for `logs` and `verifiedRequests` in conjunction with `forEach` is less appealing than the clean `map` that existed before, but I needed a way to drop the request while still pushing the log for when `request.sponsorWalletAddress !== expectedSponsorWalletAddress` and I couldn't get around the map return type constraint.

Reply:
I'd keep an `logs` array just as you use now and would use the `.filter` to filter the actionable requests.

Reply:
`.filter` worked nicely, thanks

Reply:
I prefer `.reduce` to pushing to the array as a side effect from `.filter`, but if you guys prefer the side effect approach then I'm ok with that. It's less readable to use `.reduce` but I would definitely consider it "safer"

Reply:
I implemented it with `reduce`, but particularly with prettier formatting it became very easy to miss how requests were being filtered. I'll leave as is, but this code block will be revisited when other `RequestStatus` enum variants are dropped, at which point I'll have another look if there is a way to simplify and/or use `.reduce`.

Reply:
@andreogle - if my above comment works for you, could you approve so I can merge?

Reply:
Sure - happy with that then. Thanks for taking a look 👍🏻